dev-tools/mage/common.go

```diff
@@ -913,3 +915,44 @@ func IntegrationTestEnvVars() []string {
 	}
 	return vars
 }
+
+// ReadGLIBCRequirement returns the required glibc version for a dynamically
+// linked ELF binary. The target machine must have a version equal to or
+// greater than (newer) the returned value.
+func ReadGLIBCRequirement(elfFile string) (*SemanticVersion, error) {
+	e, err := elf.Open(elfFile)
+	if err != nil {
+		return nil, err
+	}
+
+	symbols, err := e.DynamicSymbols()
+	if err != nil {
+		return nil, err
+	}
+
+	versionSet := map[SemanticVersion]struct{}{}
+	for _, sym := range symbols {
+		if strings.HasPrefix(sym.Version, "GLIBC_") {
+			semver, err := NewSemanticVersion(strings.TrimPrefix(sym.Version, "GLIBC_"))
+			if err != nil {
+				continue
+			}
+
+			versionSet[*semver] = struct{}{}
+		}
+	}
+
+	var versions []SemanticVersion
+	for ver := range versionSet {
+		versions = append(versions, ver)
+	}
+
+	sort.Slice(versions, func(i, j int) bool {
+		a := versions[i]
+		b := versions[j]
+		return a.LessThan(&b)
+	})
+
+	max := versions[len(versions)-1]
```

Comment:
I guess its impossible but this will crash for binaries that are not linked against GLIBC due to versions being empty.

Reply:
Good point. I'm adding an error if no GLIBC symbols are found.